### PR TITLE
removed c style casts and enabled the lint

### PR DESCRIPTION
**Pull request by gaaclarke**
test exempt: should have no functional change

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/engine/blob/main/docs/testing/Testing-the-engine.md
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md

### CODE REVIEW

File: impeller/entity/geometry/point_field_geometry.cc [hunk 1]
```diff
@@ -64,7 +64,7 @@ GeometryResult PointFieldGeometry::GetPositionBuffer(
 
           Point center = points_[0];
           for (auto& vertex : circle_vertices) {
-            output[offset++] = Point(center + vertex);
+            output[offset++] = static_cast<Point>(center + vertex);
```
Comment (jonahwilliams) [inline impeller/entity/geometry/point_field_geometry.cc:67]: wait, where is the cast where. both center and vertex are `Point`s

Comment (gaaclarke) [reply impeller/entity/geometry/point_field_geometry.cc:67]: Yea, I was looking at that.  Here is the error:
```
❌ Failures for clang-tidy on /Users/aaclarke/dev/engine/src/flutter/impeller/entity/geometry/point_field_geometry.cc:
/Users/aaclarke/dev/engine/src/flutter/impeller/entity/geometry/point_field_geometry.cc:67:32: error: C-style casts are discouraged; use static_cast (if needed, the cast may be redundant) [google-readability-casting,-warnings-as-errors]
   67 |             output[offset++] = Point(center + vertex);
      |                                ^~~~~
      |                                static_cast<Point>
```

I think the cast was redundant and the lint is able to convert it to what it's doing, not realize it's redundant.

Comment (jonahwilliams) [reply impeller/entity/geometry/point_field_geometry.cc:67]: oic!

Comment (gaaclarke) [reply impeller/entity/geometry/point_field_geometry.cc:67]: There are cases where it did determine they are redundant for something like `bool(false)`, it being an expression must have messed that up.

**Review by jonahwilliams — APPROVED** [review]
LGTM for great justice

**Pull request merged**